### ISSUE TITLE
CONFIG_SPIRAM_IGNORE_NOTFOUND does not work on ESP32-PICO-V3 (but works on ESP32-PICO-D4) (IDFGH-14168)

### ISSUE THREAD

**Issue by mmrein** [espressif/esp-idf#14970]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v4.4.6, v5.4.0-beta1

### Espressif SoC revision.

ESP32-PICO-V3

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

ESP-PICO-KIT-1

### Power Supply used.

USB

### What is the expected behavior?

When using `CONFIG_SPIRAM_IGNORE_NOTFOUND=y` the app should be able to boot without PSRAM. This works correctly on ESP32-PICO-D4 (no PSRAM) for example: 

```
I (5643) boot: Loaded app from partition at offset 0x20000 
I (5643) boot: Disabling RNG early entropy source... 
I (5653) quad_psram: This chip is ESP32-PICO 
E (5654) quad_psram: PSRAM ID read error: 0xffffffff, PSRAM chip not found or not supported 
E (5658) esp_psram: PSRAM enabled but initialization failed. Bailing out. 
I (5665) cpu_start: Failed to init external RAM; continuing without it. 
... app boot continues normally
```

### What is the actual behavior?

When this option is enabled on ESP32-PICO-V3 (no PSRAM) the app boot fails and ESP reboots:

```
I (607) boot: Loaded app from partition at offset 0x20000 
I (623) boot: Set actual ota_seq=1 in otadata[0] 
I (623) boot: Disabling RNG early entropy source... 
E (632) quad_psram: This chip is ESP32-PICO-V3. It does not support PSRAM (disable it in Kconfig) 
abort() was called at PC 0x40086a9c on core 0 
Backtrace: ....
...
Rebooting... 
```

### Steps to reproduce.

Using following options in sdkconfig file:

```
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
CONFIG_SPIRAM_IGNORE_NOTFOUND=y
```

### More Information.

ESP32-PICO-V3-02 works correctly as it does have PSRAM.

**Comment by Icarus113** [comment]
@mmrein have reproduced your issue, the fix will be updated here soon~

**Comment by mmrein** [comment]
Thank you for quick solve! 

I've successfully tested the fix on v4.4.6 too, just the line and file in question is this one, in case someone else needs it: https://github.com/espressif/esp-idf/blob/357290093430e41e7e3338227a61ef5162f2deed/components/esp_hw_support/port/esp32/spiram_psram.c#L839

Change to `return ESP_FAIL;`
